source/slang/slang-emit-spirv.cpp

```diff
@@ -3835,9 +3836,13 @@ struct SPIRVEmitContext
                             {
                                 switch (entryPointDecor->getProfile().getStage())
                                 {
+                                case Stage::Mesh:
+                                    // For outputs decorated with PrimitiveID it must be decorated
+                                    // with PerPrimitiveEXT to comply with Vulkan validation rules
+                                    if (m_capabilities.contains(SpvCapabilityMeshShadingEXT))
```

Comment:
This `if` seems unnecessary because if the builtin is used by a mesh shader then that capability must have been defined.

Reply:
I think we should get rid of use of NV_mesh_shader in the compiler since there is no reason to use it anymore.

Reply:
Ok do you want me to create a new bug for removing support for NV_mesh_shader? In that case I can remove this check and just add a comment.